### PR TITLE
Fully functional HTTP client with tests :D

### PR DESCRIPTION
**Pull request by guillermoares**
In this PR:

- The HTTP client is now fully functional.
- Test suit for said client. Easily applicable to the gRPC client in the future.
- Wrote a small README. A lo of room for improvement, though.
- Added *javadocs* to the `EntityHandler` interface.
- Added package.json script to generate the gRPC files. Also added this to the *build* script.
- Explicitly typed the client's exposed interface.

### CODE REVIEW

File: src/__tests__/metadata-api-client.test.ts [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+	SAMPLE_PUBLISHERS,
+	SAMPLE_SOURCES,
+	TARGET_HOST,
+} from './constants';
+
+describe('MetadataAPIClient', () => {
+	defineTestsForImplementation('HTTP', () =>
+		MetadataAPIClient.UsingHTTP(TARGET_HOST),
+	);
+	// defineTestsForImplementation("gRPC", () => MetadataAPIClient.UsingGRPC(HOST))
```
Comment (guillermoares) [inline src/__tests__/metadata-api-client.test.ts]: Once gRPC is available, uncommenting this line will make the tests run also for the gRPC client.

File: package.json [hunk 2]
```diff
@@ -5,7 +5,8 @@
   "main": "dist/index.js",
   "types": "dist/index.d.ts",
   "scripts": {
-    "build": "npx tsc",
+    "protoc": "cd proto && sh ./generate-protoc-code.sh",
```
Comment (skumar221) [inline package.json]: Does this behave in docker land?

File: proto/generate-protoc-code.sh [hunk 3]
```diff
@@ -1,15 +1,14 @@
 #!/usr/bin/env bash
 
-OUT_DIR="./generated"
-# Path to this plugin
+# Path to the grpc_node_plugin
+PROTOC_GEN_GRPC_PATH="./../node_modules/.bin/grpc_tools_node_protoc_plugin"
+# Path to TS plugin
 PROTOC_GEN_TS_PATH="./../node_modules/.bin/protoc-gen-ts"
+OUT_DIR="./generated"
```
Comment (skumar221) [inline proto/generate-protoc-code.sh]: Does this need to be copied in the docker file?

File: package.json (hunk 2 above)
Comment (skumar221) [reply package.json]: Disregard!  I thought this was omnom.  Sorry!

File: proto/generate-protoc-code.sh (hunk 3 above)
Comment (skumar221) [reply proto/generate-protoc-code.sh]: Disregard!

File: src/model/entity-handlers/http-entity-handler.ts [hunk 4]
```diff
@@ -11,24 +13,92 @@ export class HTTPEntityHandler extends EntityHandler {
...
+	async getAll() {
+		return await this._requestHandlingServerErrors({
+			method: 'GET',
+			uri: this._entityEndpoint(),
+			json: true,
+		});
+	}
+
+	async update(entity) {
+		await this.get(entity.id);
```
[the remark this answers is not in the document]
Comment (guillermoares) [reply src/model/entity-handlers/http-entity-handler.ts]: Yes! That's the idea. Thing is the server actually only does *upsert*, so without it `create` and `update` are exactly the same thing.

**Pull request merged**